lib/shoulda/matchers/active_record/have_db_index_matcher.rb

```diff
@@ -65,6 +65,56 @@ module ActiveRecord
       #
       # @return [HaveDbIndexMatcher]
       #
+      # has an index and an array of columns
```

Comment:
Instead of putting this at the end, I would put this right after the first section, before `#### Qualifiers`.

Additionally, it'd be nice to flesh out the first line here so it introduces _this_ section a little more naturally.